src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -0,0 +1,1511 @@
+				waiting_access_point = false; // Next event is the ctxLink to AP event
+			} else {
+				DEBUG_WARN("APP_WIFI_CB[%d]: Connected to AP\r\n", msg_type);
+				g_wifi_connected = true;
+			}
+		} else if (wifi_state->u8CurrState == M2M_WIFI_DISCONNECTED) {
+			DEBUG_WARN("APP_WIFI_CB[%d]: Disconnected from AP\r\n", msg_type);
+			g_wifi_connected = false;
+		} else {
+			DEBUG_WARN("APP_WIFI_CB[%d]: Unknown WiFi state change\r\n", msg_type);
+		}
+		break;
+	}
+
+	case M2M_WIFI_IP_ADDRESS_ASSIGNED_EVENT: {
+		t_wifiEventData *wifi_event_data = (t_wifiEventData *)msg;
+		if (wifi_event_data != NULL) {
+			ip_address_assigned = true;
+		}
+		ip_address_assigned = true;
+		break;
+	}
+
+	case M2M_WIFI_WPS_EVENT: {
+		tstrM2MWPSInfo *wifi_wps_info = (tstrM2MWPSInfo *)msg;
+		DEBUG_WARN("Wi-Fi request WPS\r\n");
+		DEBUG_WARN("SSID : %s, authtyp : %d pw : %s\n", wifi_wps_info->au8SSID, wifi_wps_info->u8AuthType,
+			wifi_wps_info->au8PSK);
+		if (wifi_wps_info->u8AuthType == 0) {
+			DEBUG_WARN("WPS is not enabled OR Timedout\r\n");
+			/*
+					WPS monitor timeout.
+				 */
+			m2m_wifi_wps_disable();
+			wps_active = false;
+		} else {
+			DEBUG_WARN("Request Wi-Fi connect\r\n");
+			m2m_wifi_connect_sc((char *)wifi_wps_info->au8SSID, strlen((char *)wifi_wps_info->au8SSID),
+				wifi_wps_info->u8AuthType, wifi_wps_info->au8PSK, wifi_wps_info->u8Ch);
+		}
+
+		break;
+	}
+	case M2M_WIFI_PROVISION_INFO_EVENT: {
+		tstrM2MProvisionInfo *provision_info = (tstrM2MProvisionInfo *)msg;
+		if (provision_info->u8Status == M2M_SUCCESS) {
+			m2m_wifi_connect_sc((char *)provision_info->au8SSID, strlen((char const *)provision_info->au8SSID),
+				provision_info->u8SecType, provision_info->au8Password, M2M_WIFI_CH_ALL);
+		} else {
+			m2m_wifi_stop_provision_mode();
+		}
+		break;
+	}
+	case M2M_WIFI_DEFAULT_CONNNECT_EVENT: {
+		//tstrM2MDefaultConnResp *pDefaultConnResp = (tstrM2MDefaultConnResp *) msg;
+		DEBUG_WARN("APP_WIFI_CB[%d]: Un-implemented state\r\n", msg_type);
+		break;
+	}
+		/* Unused states. Can be implemented if needed  */
+	case M2M_WIFI_CONN_INFO_RESPONSE_EVENT:
+	case M2M_WIFI_SCAN_DONE_EVENT:
+	case M2M_WIFI_SCAN_RESULT_EVENT:
+	case M2M_WIFI_SYS_TIME_EVENT:
+	case M2M_WIFI_PRNG_EVENT:
+	case M2M_WIFI_IP_CONFLICT_EVENT:
+	case M2M_WIFI_INVALID_WIFI_EVENT:
+	case M2M_WIFI_RSSI_EVENT: {
+		DEBUG_WARN("APP_WIFI_CB[%d]: Un-implemented state\r\n", msg_type);
+		break;
+	}
+
+	default: {
+		DEBUG_WARN("APP_WIFI_CB[%d]: Unknown WiFi state\r\n", msg_type);
+		break;
+	}
+	}
+}
+
+bool is_driver_init_complete(void)
+{
+	bool res = driver_init_complete;
+	driver_init_complete = false;
+	return res;
+}
+
+bool is_wifi_connected(void)
+{
+	bool res = g_wifi_connected;
+	// no need to reset flag "g_wifi_connected" to false. Event will do that.
+	return res;
+}
+
+bool is_ip_address_assigned(void)
+{
+	bool res = ip_address_assigned;
+	ip_address_assigned = false;
+	return res;
+}
+
+void handle_socket_bind_event(SOCKET *lpSocket, bool *lpRunningState)
```

Comment:
Nomenclature - and same for the next few functions. `lower_camel_case` with non-type-suffixed names please.

Reply:
Fixed

Reply:
On the next line?

Reply:
Yes, indeed.